zebra/if_netlink.c

```diff
@@ -350,11 +350,38 @@ static void netlink_vrf_change(struct nlmsghdr *h, struct rtattr *tb,
 			}
 		}
 
-		/*
-		 * vrf_get is implied creation if it does not exist
+		/*Treat VRF add for existing vrf as update
+		 * Update VRF ID and also update in VRF ID table
 		 */
-		vrf = vrf_get((vrf_id_t)ifi->ifi_index,
-			      name); // It would create vrf
+		if (name)
+			vrf = vrf_lookup_by_name(name);
+		if (vrf && (vrf_id_t)ifi->ifi_index != VRF_UNKNOWN
+		    && vrf->vrf_id != VRF_UNKNOWN
+		    && vrf->vrf_id != (vrf_id_t)ifi->ifi_index) {
+			zlog_info(
+				"Vrf Update event: %s old id: %u table:%u new id: %u table: %u",
+				name, vrf->vrf_id,
+				((struct zebra_vrf *)vrf->info)->table_id,
+				(vrf_id_t)ifi->ifi_index, nl_table_id);
+
+			/*Disable the vrf to simulate implicit delete
+			 * so that all stale routes are deleted
+			 * This vrf will be enabled down the line*/
+			vrf_disable(vrf);
+
+
+			RB_REMOVE(vrf_id_head, &vrfs_by_id, vrf);
```

Comment:
This code seems a bit overly complex to be in netlink dataplane logic. Can we not abstract this to be a vrf_update() call or something?

Reply:
I have done the changes.

Reply:
You haven't done this. Maybe I was not clear in my working... my apologies. I meant this code should be abstracting out of the if_netlink.c file and handled similarly to how is being done for `vrf_delete()` and `vrf_get()`  you should create a new `vrf_update()` or something that can handle both add/update and call that in lib/vrf.c

Reply:
@sworleys  I have done the changes , please approve.